documentation/binding-data/tutorial-flow-data-provider.asciidoc

```diff
@@ -53,535 +112,429 @@ grid.setItems(
+Alternatively, you can use more fine-grained APIs in the `DataView` to update just a portion of the dataset.
 
-*Example*: Using the `withConfigurableFilter` method to create a data provider wrapper.
+For example, you can modify a displayed item and notify `Grid` about the updates to the item through the `DataView` API.
 
 [source, java]
 ----
-DataProvider<Employee, String> employeeProvider =
-        getEmployeeProvider();
+Person person = new Person();
+person.setName("Jorma");
+person.setEmail("old@gmail.com");
 
-ConfigurableFilterDataProvider<Employee, Void, String>
-    wrapper = employeeProvider.withConfigurableFilter();
+GridListDataView<Person> gridDataView = grid.setItems(person);
 
-Grid<Employee> grid = new Grid<>();
-grid.setDataProvider(wrapper);
-grid.addColumn(Employee::getName).setHeader("Name");
+Button modify = new Button("Modify data", e -> {
+    person.setEmail("new@gmail.com");
 
-searchField.addValueChangeListener(event -> {
-    String filter = event.getValue();
-    if (filter.trim().isEmpty()) {
-        // null disables filtering
-        filter = null;
-    }
-
-    wrapper.setFilter(filter);
+    // The component shows the old email until notified of changes
+    gridDataView.refreshItem(person);
 });
 ----
 
-* The filter type of the `wrapper` instance is `Void`. This means that the data provider does not support further filtering through the query. It is therefore not possible to use this data provider with a combo box.
-
-=== Refreshing Data from a Backend Service
+If you are mutating the data set bound to a component like grid, you need to reset the data binding.
 
-`DataProvider` has two methods, `refreshAll` and `refreshItems`, that you can use to ensure that backend changes reflect in all parts of you application. 
-
-Whether refreshing is required depends on your implementation and environment. Spring Data, for example, gives new instances with every request, and changes to the repository make old instances of the same object "stale". In cases similar to this, you should inform interested components by calling `dataProvider.refreshItem(newInstance)`. This works out of the box, if your beans have equals and hashCode implementations that check if the objects represent the same data. Since this is not always the case, when using `CallbackDataProvider` you can give it a `ValueProvider` that will provide a stable ID for the data objects. This is usually a method reference, for example `Person::getId`.
-
-*Example*: `PersonService` interface with an update method that returns a new instance of the item. _Other functionality is omitted._
-
-[source, java]
-----
-public interface PersonService {
-    Person save(Person person);
-}
-----
+If you have bound a mutable `List` to your component, you can alternatively use helper methods in the list data view to add or remove items or obtain item count by hooking to item count change event or request the item count directly.
```

Comment:
Methods for adding, removing and assigning a listener to component are listed in example below. Reset is made by `setItems`

Reply:
The question is asked not from my side personally.
Every question which I ask is asked from the reader of the text.
Normally the clarification should be done as a fix in the text and not in the comment.

Github unfortunately doesn't give me an obvious way to attach a comment to the specific word.

The question was about `you need to reset the data binding`.
Now when I read it one more time the sentence says to me that the method for reset is `setItems`.

I expected here some API which allows me reset data in the component without completely reset the whole data.
So it was confusing for me.

But then `If you are mutating the data set bound to a component like grid`  : is the only way to update the data in the component reset the whole data ?
Can I make changes in one bean and reset something (data provider, components, etc.) without setting the whole data again ?
Is there such section in the docs ?

Reply:
I propose to remove `If you are mutating the data set bound to a component like grid, you need to
reset the data binding` because it's a bit confusing. Don't know what does the `reset` mean in this context. I would better remove it at this point and ask Matti latter, what feature he wanted to emphasise. 

`by using a one of overloaded `setItems()` methods` is apparently incorrect because it just binds a new data set to component, not a reset. 

> Can I make changes in one bean and reset something (data provider, components, etc.) without setting the whole data again ?

So, the only method to refresh a particular item in data set (after you changed some fields) - to call `dataView.refreshItem(item)`. It just refreshes only one item, like the old style `DataProvider::refreshItem` did. You can also use the `DataProvider` API, but the idea is to gradually replace the `DataProvider` with `DataView`, that's why we don't mention it here.

This is described here (I changed a wording a bit to emphasise only one item is refreshed):
```
For example, you can modify the field(s) of a displayed item and notify 
`Grid` about the updates to the item through the `DataView` API. This would modify
only one particular item, not the whole data set.
```
Sorry, I stuck here, but really don't see how more I could improve this section.